server/util.go

```diff
@@ -412,3 +416,46 @@ func openTraceWriter(traceWriterFile string) (w io.Writer, err error) {
 		0o666,
 	)
 }
+
+// DefaultBaseappOptions returns the default baseapp options provided by the Cosmos SDK
+func DefaultBaseappOptions(appOpts types.AppOptions) []func(*baseapp.BaseApp) {
```

Comment:
Differences between these `DefaultBaseappOptions` and the provenance ones:
1. When reading the `FlagIAVLCacheSize` value, we enforce a default. As long as we're loading the defaults before reading config files, though, I don't think we need to continue doing that.
2. We create the snapshots directory if it doesn't exist. I seem to remember those lines being a pain-point with the most recent upgrade, though due to getting the home dir from its flag. At the very least, we'll need to update the `provenance` persistent pre-run stuff to set the home flag in viper so that it's got the correct default.

Reply:
I double checked the snapshots dir thing, and it's fine if we don't explicitly create it anymore.

Reply:
I tested the `FlagIAVLCacheSize` thing too. I added a log line to output what appOpt.Get returns for it, and did a `make run-config`, edited app.toml to remove that entry, and then `make run`. The log entry indicates a value of `781250` as expected (hoped?).

Reply:
Summary:

It's safe for us to switch to using this function.